pkg/k8shandler/logstore_test.go

```diff
@@ -347,3 +348,77 @@ func TestDifferenceFoundWhenNodeCountExceeds4(t *testing.T) {
 		t.Errorf("Expected that difference would be found due to node count change")
 	}
 }
+
+func TestGenUUIDPreservedWhenNodeCountExceeds4(t *testing.T) {
+	cluster := &logging.ClusterLogging{
+		Spec: logging.ClusterLoggingSpec{
+			LogStore: logging.LogStoreSpec{
+				Type: "elasticsearch",
+				ElasticsearchSpec: logging.ElasticsearchSpec{
+					NodeCount: 3,
+				},
+			},
+		},
+	}
+	elasticsearchCR := newElasticsearchCR(cluster, "test-app-name")
+	dataUUID := esutils.GenerateUUID()
+	elasticsearchCR.Spec.Nodes[0].GenUUID = &dataUUID
+
+	cluster = &logging.ClusterLogging{
```

Comment:
I don't recall if this is saved via pointer or value but can you reduce the lines, reusing the `cluster` and simply bump the node count so you don't need to redefine everything?

Reply:
I probably could, this is the same as our other functions though... tech debt?